test/endpoints/favourite_filters/test_save_filter.py

```diff
@@ -0,0 +1,32 @@
+from bson import ObjectId
+from fastapi.testclient import TestClient
+
+from test.endpoints.favourite_filters.conftest import FavouriteFilter
+
+
+class TestSaveFilter:
+    def test_save_filter(
+        self,
+        test_app: TestClient,
+        login_and_get_token,
+    ):
+        name = "Test Create Filter"
+        filter_content = "PM-201-HJ-CRY-FLOW > 10"
+
+        test_response = test_app.post(
+            f"/users/filters?name={name}&filter={filter_content}",
```

Comment:
We're only testing this when `name` and `filter` is set, when (as mentioned on the endpoint) they will also default to `""` if not set, resulting in (e.g.) `[{'_id': ObjectId('6655fb00434b426cbaed6370'), 'filter': '', 'name': ''}]`. If this is indeed intentional, then we should test that this functionality is correct with an additional test case where the query params are not set.

Reply:
No, this functionality isn't intended - I've added two test cases to test the intended functionality in 1c493c6e455dcbd9a900d7eb689c5381bc6fc603